omod/src/main/java/org/openmrs/module/locationbasedaccess/fragment/controller/field/LocationsFragmentController.java

```diff
@@ -0,0 +1,39 @@
+package org.openmrs.module.locationbasedaccess.fragment.controller.field;
+
+import org.openmrs.Location;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Fields and actions to provide registrationInfo view.
+ *
+ * @author Romain Buisson
+ */
+public class LocationsFragmentController {
+
+    public void controller(FragmentModel model,
+                           HttpSession session) {
+
+        // Fetch the current logged in attribute from the session
+        String LOCATION_SESSION_ATTRIBUTE = UiSessionContext.LOCATION_SESSION_ATTRIBUTE;
+        Integer sessionLocationId = (Integer) session.getAttribute(LOCATION_SESSION_ATTRIBUTE);
```

Comment:
What is the use of the above logic? Aren't you just interested in getting the list of locations?

Reply:
Nope, Here I'm getting all the locations list and select the session location from that list defaultly.

Eg : If the session location is A, then the location selector will be selected with A. Then user can change to other locations using this selector.